src/librustdoc/clean/mod.rs

```diff
@@ -327,17 +327,27 @@ impl Item {
         }
     }
 
-    pub fn stability_class(&self) -> String {
-        self.stability.as_ref().map(|ref s| {
-            let mut base = match s.level {
-                stability::Unstable => "unstable".to_string(),
-                stability::Stable => String::new(),
-            };
-            if !s.deprecated_since.is_empty() {
-                base.push_str(" deprecated");
+    pub fn stability_class(&self) -> Option<String> {
+        match self.stability {
```

Comment:
I *believe* (I'm no expert either) this would be more idiomatic:

```
self.stability.as_ref().and_then(|ref s| {
// ...
})
```

one level less of indentation, no need to map None to None. I made a tiny playground script to make sure they both mean the same thing if you wanna take a look https://is.gd/5fSbZf